Sources/Sentry/SentryFileManager.m

```diff
@@ -158,7 +158,19 @@ - (void)createPathsWithOptions:(SentryOptions *)options
     SENTRY_LOG_DEBUG(@"SentryFileManager.cachePath: %@", cachePath);
 
     self.basePath = [cachePath stringByAppendingPathComponent:@"io.sentry"];
-    self.sentryPath = [self.basePath stringByAppendingPathComponent:[options.parsedDsn getHash]];
+
+    NSString *_Nullable nullableDsnHash = [options.parsedDsn getHash];
+    if (nullableDsnHash == nil) {
+        SENTRY_LOG_FATAL(@"No DSN provided, using base path for envelopes: %@", self.basePath);
+    }
+    // We decided against changing the `sentryPath` and use a null fallback instead, because the
+    // impact of changing the base path can result in critical issues.
```

Comment:
`m`: I would add that this is broken for a very long time already and we plan on fixing this in an GH issue. You could link the issue.